tests/10apidoc/34room-messages.pl

```diff
@@ -44,10 +44,17 @@ sub matrix_send_room_message
 
    my $type = $opts{type} // "m.room.message";
 
-   do_request_json_for( $user,
-      method => "POST",
-      uri    => "/api/v1/rooms/$room_id/send/$type",
+   my $method = "POST";
+   my $uri = "/api/v1/rooms/$room_id/send/$type";
+
+   if (defined $opts{txn_id}) {
+      $method = "PUT";
+      $uri = "$uri/${\ $opts{txn_id} }";
```

Comment:
This doesn't need the `${\...}` interpolation hack; hash derefs interpolate nicely. It's just method calls that don't.

    $uri = "$uri/$opts{txn_id}";